src/main/kotlin/me/partlysanestudios/partlysaneskies/data/api/RequestsManager.kt

```diff
@@ -0,0 +1,97 @@
+package me.partlysanestudios.partlysaneskies.data.api
+
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies
+import me.partlysanestudios.partlysaneskies.utils.MathUtils
+import java.io.IOException
+import java.util.*
+
+object RequestsManager {
+
+    private val requestsQueue: Queue<Request> = LinkedList()
+    private var lastRequestTime: Long = 0
+
+
+    val thread = Thread({
+        while(true) {
```

Comment:
Oh god

Reply:
Have a better solution?

Reply:
Honestly no

Reply:
While true it is

Reply:
I will add a sleep at the end of the thread